### PR TITLE
Backport #11625, #15101

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

File: src/qt/dash.cpp [hunk 1]
```diff
@@ -501,24 +409,27 @@ void BitcoinApplication::initializeResult(bool success)
         } else {
             window->showMinimized();
         }
-        Q_EMIT splashFinished(window);
+        Q_EMIT splashFinished();
```
Comment (UdjinM6) [inline src/qt/dash.cpp:372]: 11625: this should be a part of 14854

File: src/qt/dash.cpp [hunk 2]
```diff
@@ -501,24 +409,27 @@ void BitcoinApplication::initializeResult(bool success)
...
+            connect(paymentServer, &PaymentServer::message, [this](const QString& title, const QString& message, unsigned int style) {
+                window->message(title, message, style);
+            });
+            QTimer::singleShot(100, paymentServer, &PaymentServer::uiReady);
+        }
 #endif
         pollShutdownTimer->start(200);
     } else {
-        Q_EMIT splashFinished(window); // Make sure splash screen doesn't stick around during shutdown
+        Q_EMIT splashFinished(); // Make sure splash screen doesn't stick around during shutdown
```
Comment (UdjinM6) [inline src/qt/dash.cpp:392]: 11625: this should be a part of 14854

File: src/qt/dash.h [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    void handleRunawayException(const QString &message);
+    void addWallet(WalletModel* walletModel);
+    void removeWallet();
+
+Q_SIGNALS:
+    void requestedInitialize();
+    void requestedRestart(QStringList args);
+    void requestedShutdown();
+    void stopThread();
+    void splashFinished();
```
Comment (UdjinM6) [inline src/qt/dash.h:100]: 11625:
```suggestion
    void splashFinished(QWidget *window);
````
and then drop `QWidget *window` in 14854

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by UdjinM6** [comment]
14854 was merged via #4412 already

**Comment by PastaPastaPasta** [comment]
Okay, should be ready

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**